Dockerfile-alternate

```diff
@@ -0,0 +1,3 @@
+FROM prairielearn/prairielearn
```

Comment:
This removes all the installation stuff and uses the latest PrairieLearn image directly, speeds up starting the production server by quite a bit. If we don't want this I can remove it.

Reply:
Could we just remove the `build` section from the `pl` service in the Docker Compose config and instead rely on the `prairielearn/prairielearn` image that's published to Docker Hub? I think that should be the default case.

Reply:
Independent users of PL may want to pin their server version (suppose PL makes a major breaking update or something, and they're not ready for it). If there's a rebuild involved here they can specify a different upstream image based on its hash or version tag. Would that be a reasonable use case?

Reply:
On the other hand, this may encourage someone to carelessly downgrade the image, and I don't know if PL would handle migrations or DB breakage going in the other direction.

Reply:
@echuber2 I agree with this, also if they would like to install some new dependencies on top of the PrairieLearn image, they could do so easily through the `Docker-alternate` file.

i.e. adding the line to the file:
`RUN pip install [new dependency]`

Reply:
@echuber2 if pinning is something that folks want support for, we can start publishing images tagged by the Git hash of the commit it's built from. And if we do that, we don't need a custom Dockerfile, they could just specify: `image: prairielearn/prairielearn:GIT_HASH_HERE` in their Docker Compose config.

@Prostigus candidly, I don't think we're interested in officially supporting the use case of including additional dependencies. That would make it more difficult for anyone trying to do local course development, i.e. they'd have to also build that custom image locally instead of just doing `docker run prairielearn/prairielearn`. Of course, we can't stop you from doing that if you want to, but I don't want that potential complexity to leak into our official docs.

Reply:
I'd agree with tagging the images with the commit. (I've had to pin versions of other upstream projects before when they did something breaking.)

Reply:
What would happen if somebody ran the latest build of PL and it did migrations, then they tried to roll back to a pinned previous version? Would PL be able to handle it gracefully?

Reply:
We don't test rollbacks of code after applying database migrations, nor do we make any guarantees about what would happen if you tried that.

You're welcome to open an issue or PR for tagging images, but you can even pin to specific versions without making any changes to our tagging strategy, e.g. `prairielearn/prairielearn@sha256:0238c426cfdc347d972b34eafef584207cb7a1b1ffceb72885e1ad33c5807f62`. You can find the image SHA with `docker inspect prairielearn/prairielearn`.

Reply:
Agreed, I've removed `build` and changed `prairielearn/prairielearn:local` to `prairielearn/prairielearn:latest`